internals/overlord/servstate/manager_test.go

```diff
@@ -547,239 +600,14 @@ func (s *S) TestStartFastExitCommand(c *C) {
-	c.Assert(layer.Order, Equals, 1)
-	c.Assert(planYAML(c, s.manager), Equals, `
-services:
-    svc1:
-        override: replace
-        command: /bin/sh
-`[1:])
-	s.planLayersHasLen(c, s.manager, 1)
-
-	// Combine layer with different label should just append.
-	layer = parseLayer(c, 0, "label2", `
-services:
-    svc2:
-        override: replace
-        command: /bin/foo
-`)
-	err = s.manager.CombineLayer(layer)
-	c.Assert(err, IsNil)
-	c.Assert(layer.Order, Equals, 2)
-	c.Assert(planYAML(c, s.manager), Equals, `
-services:
-    svc1:
-        override: replace
-        command: /bin/sh
-    svc2:
-        override: replace
-        command: /bin/foo
-`[1:])
-	s.planLayersHasLen(c, s.manager, 2)
-
-	// Combine layer with first layer.
-	layer = parseLayer(c, 0, "label1", `
-services:
-    svc1:
-        override: replace
-        command: /bin/bash
-`)
-	err = s.manager.CombineLayer(layer)
-	c.Assert(err, IsNil)
-	c.Assert(layer.Order, Equals, 1)
-	c.Assert(planYAML(c, s.manager), Equals, `
-services:
-    svc1:
-        override: replace
-        command: /bin/bash
-    svc2:
-        override: replace
-        command: /bin/foo
-`[1:])
-	s.planLayersHasLen(c, s.manager, 2)
-
-	// Combine layer with second layer.
-	layer = parseLayer(c, 0, "label2", `
-services:
-    svc2:
-        override: replace
-        command: /bin/bar
-`)
-	err = s.manager.CombineLayer(layer)
-	c.Assert(err, IsNil)
-	c.Assert(layer.Order, Equals, 2)
-	c.Assert(planYAML(c, s.manager), Equals, `
-services:
-    svc1:
-        override: replace
-        command: /bin/bash
-    svc2:
-        override: replace
-        command: /bin/bar
-`[1:])
-	s.planLayersHasLen(c, s.manager, 2)
-
-	// One last append for good measure.
-	layer = parseLayer(c, 0, "label3", `
-services:
-    svc1:
-        override: replace
-        command: /bin/a
-    svc2:
-        override: replace
-        command: /bin/b
-`)
-	err = s.manager.CombineLayer(layer)
-	c.Assert(err, IsNil)
-	c.Assert(layer.Order, Equals, 3)
-	c.Assert(planYAML(c, s.manager), Equals, `
-services:
-    svc1:
-        override: replace
-        command: /bin/a
-    svc2:
-        override: replace
-        command: /bin/b
-`[1:])
-	s.planLayersHasLen(c, s.manager, 3)
-
-	// Make sure that layer validation is happening.
-	layer, err = plan.ParseLayer(0, "label4", []byte(`
-checks:
-    bad-check:
-        override: replace
-        level: invalid
-        tcp:
-            port: 8080
-`))
-	c.Check(err, ErrorMatches, `(?s).*plan check.*must be "alive" or "ready".*`)
-}
-
+// TestSetServiceArgs does not involve the service manager plan snapshot as
+// the plan management is no longer part of the service manager. However,
+// each manager still needs the ability to provide a way to create a custom
+// layer that can be used to mutate the plan state. This test focusses on
+// testing the mutation logic of the service, and not how the changes are
+// applied to planstate.
 func (s *S) TestSetServiceArgs(c *C) {
```

Comment:
Removed parts of the test that is not strictly concerned with the Service Manager Code. The Plan Manager will have enough tests about layering, appending and combining and checking the end marshal result.

Reply:
Removed.